test/spoof/spoof.go

```diff
@@ -134,10 +135,11 @@ func New(
 	}
 
 	sc := SpoofingClient{
-		Client:          &http.Client{Transport: roundTripper},
-		RequestInterval: requestInterval,
-		RequestTimeout:  RequestTimeout,
-		Logf:            logf,
+		Client:           &http.Client{Transport: roundTripper},
+		RequestInterval:  requestInterval,
+		RequestTimeout:   RequestTimeout,
+		RetryOnAllErrors: false,
```

Comment:
Where is it going to be set to `true`?

Reply:
It would be in the serving side. The prober for upgrade test. Somewhere around 
https://github.com/knative/serving/blob/master/test/prober.go#L177

Reply:
This needs to be properly guarded for thread access then I think.
Unless it's set in the constructor and immutable for the life of the client (then it should be set in the `New` and made private).